src/battle_main.c

```diff
@@ -3089,6 +3091,7 @@ void SwitchInClearSetData(void)
     gLastResultingMoves[gActiveBattler] = 0;
     gLastPrintedMoves[gActiveBattler] = 0;
     gLastHitBy[gActiveBattler] = 0xFF;
+    gStoredHealingWish &= ~gBitTable[GetBattlerPosition(gActiveBattler) + MAX_BATTLERS_COUNT + 1];
```

Comment:
please move gStoredHealingWish to gBattleStruct. Also, why are you adding ` MAX_BATTLERS_COUNT + 1` to the battler id? What is the intended functionality here? if you're trying to just set a bit for the given battler, you just need `gBitTable[battlerId]`

Its used correctly below, e.g. `gStoredHealingWish & gBitTable[GetBattlerPosition(gActiveBattler)]`, so is setting a separate flag?

Reply:
This involved a janky workaround to try to bypass the "Use next Pokemon?" prompt in wild battles when using Healing Wish which I forgot I had made. Considering nothing similar exists for Explosion or other suicide moves, I just removed it and used the bits to make Lunar Dance work properly (there were probably some bugs with messages if Lunar Dance was used by battlers other than those in position 0).